src/orange/transform/detail/TransformTransformer.hh

```diff
@@ -0,0 +1,114 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/transform/detail/TransformTransformer.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <variant>
+
+#include "orange/MatrixUtils.hh"
+#include "orange/Types.hh"
+
+#include "../Transformation.hh"
+#include "../Translation.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Apply a transformation to another transformation.
+ *
+ * An instance of this class applies a daughter-to-parent translation to
+ * another transformation: the translation needed to relocate a "lower"
+ * universe to a new coordinate system in the "higher" universe.
+ *
+ * The operation returns a new transform defined
+ * \f[
+   \mathbf{T}' = \mathbf{T}_L \mathbf{T}_R
+ * \f]
+ * where T is the argument, \f$\mathbf{T}_L\f$ is the constructor argument (the
+ * operator itself), and the result \f$\mathbf{T}'\f$ is the return value.
+ * The resulting transform is:
+ * \f[
+   \mathbf{x}' = \mathbf{T}_L\mathbf{x}
```

Comment:
What is `x` here? Could you double check this section? Lines 47-53 are easy to follow, but 38-45 seem like there might be an error somewhere.